<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN">

<html>

<head>
<link type="text/css" rel="stylesheet" href="style.css">
<title>Peter Carbonetto</title>
</head>

<body>
<center>
<br>
<table align="center" border=0 width=540 cellspacing=0
cellpadding=0>
<tr><td valign=top>
<h2 style="font-size:16px;"><b>Peter Carbonetto</b>, Ph.D.<br>
Research Assistant Professor<br>
<a href="https://genes.uchicago.edu">Dept. of Human Genetics</a><br>
<a href="https://www.uchicago.edu">University of Chicago</a></h2>
<p><b>pcarbo -at- uchicago -dot- edu</b></p> 

<p><h2>I'm a Research Assistant Professor at the
<a href="https://www.uchicago.edu">University of Chicago</a>. 
Previously, I was a Staff Scientist at <a href="">Ancestry</a>; a postdoc and 
<a href="http://www.hfsp.org">HFSP fellow</a> working with 
<a href="http://stephenslab.uchicago.edu">Matthew Stephens</a> and
<a href="http://palmerlab.org">Abraham Palmer</a> in the
<a href="http://genes.uchicago.edu">Dept. of Human Genetics</a>
at the <a href="http://www.uchicago.edu">University of
Chicago</a>; a graduate student with 
<a href="http://www.cs.ubc.ca/spider/nando">Nando de Freitas</a>
in the <a href="http://www.cs.ubc.ca/labs/lci">Laboratory for
Computational Intelligence</a> at <a href="http://www.ubc.ca">UBC.</a>
My work is focused on developing new quantitative approaches 
to advance the study of global genetic variation in health and 
disease.</h2></p>

</td>

<td>&nbsp;&nbsp;</td>

<td valign=top><table width=144 align=center>
<tr><td><p class=img><img src="napa.jpg" 
height=189 width=160 alt="Peter" hspace=4 vspace=4></p>

<p><b>Mailing address:</b><br> 
CLSC 414 / 920 E 58th St / Chicago, IL 60637</p>

</td></tr></table></td></tr>

<tr>
<td colspan=3 align=center>
<h5>
<a href="http://github.com/pcarbo">GitHub</a> |
<a href="https://profiles.uchicago.edu/profiles/display/37244">Research
Network Profile</a> |
<a href="http://scholar.google.com/citations?user=ghacqrQAAAAJ">Google 
Scholar</a> |
<a href="http://www.linkedin.com/pub/peter-carbonetto/59/8ab/7a">LinkedIn</a>
</h5>
</td>
</tr>

<tr><td colspan=3>

<a name=pubs></a><h1>Publications</h1>

<p class=pub>
W. R. P. Denault, K. Tayeb, P. Carbonetto, J. Willwerscheid and
M. Stephens.
<a href="https://openreview.net/forum?id=OrmLtoFF60">Covariate-moderated 
Empirical Bayes Matrix Factorization</a>
Neural Information Processing Systems 38,
December 2025.</p>

<p class=pub>
Jason Willwerscheid, Peter Carbonetto and Matthew Stephens. 
<a href="https://doi.org/10.18637/jss.v114.i03">ebnm: an R
Package for solving the empirical Bayes normal means problem using a
variety of prior families.</a>
<a href="https://www.jstatsoft.org/"><b>Journal of Statistical 
Software</b></a>, volume 114, 2025. 
<a href="https://github.com/stephenslab/ebnm">R package</a>
</p>

<p class=pub>
E. E. Thompson, X. Zhong, P. Carbonetto, A. Morin, J. Willwerscheid,
C. M. Visness, L. B. Bacharier, M. Kattan, G. T. O'Connor,
K. Rivera-Spoljaric, R. A. Wood, D. R. Gold, G. K. K. Hershey,
C. C. Johnson, R. L. Miller, C. M. Seroogy, E. M. Zoratti,
P. J. Gergen, A. M. Levin, M. C. Altman, T. Hartert, M. Stephens,
D. J. Jackson, J. E. Gern, C. G. McKennan and C. Ober.
<a href="https://doi.org/10.1016/j.ajhg.2025.05.006">Genetic contributions 
to epigenetic-defined endotypes of allergic phenotypes in children.</a>
<a href="https://www.cell.com/ajhg/"><b>American Journal of Human 
Genetics</b></a>, volume 112, July 2025.
</p>

<p class=pub>
Yusha Liu, Peter Carbonetto, Jason Willwerscheid, Scott A. Oakes, Kay F. 
Macleod and Matthew Stephens. 
<a href="https://doi.org/10.1038/s41588-024-01997-z">Dissecting 
tumor transcriptional heterogeneity from single-cell RNA-seq data by 
generalized binary covariance decomposition.</a>
<a href="https://www.nature.com/ng/"><b>Nature Genetics</b></a>, volume 57,
January 2025.
<a href="https://github.com/stephenslab/gbcd">R package</a>
</p>

<p class=pub>
Eric Weine, Peter Carbonetto and Matthew Stephens.
<a href="https://doi.org/10.1093/bioinformatics/btae494">Accelerated
dimensionality reduction of single-cell RNA sequencing data with
fastglmpca.</a>
<a href="https://academic.oup.com/bioinformatics"><b>Bioinformatics</b></a>,
volume 40, August 2024.
<a href="https://github.com/stephenslab/fastglmpca">R package</a></p>  

<p class=pub>
Yusha Liu, Peter Carbonetto, Michihiro Takahama, Adam Gruenbaum,
Dongyue Xie, Nicolas Chevrier and Matthew Stephens.
<a href="https://doi.org/10.1214/24-AOAS1894">A flexible model for
correlated count data, with application to multicondition differential
expression analyses of single-cell RNA sequencing data.</a>
<a href="https://projecteuclid.org/journals/annals-of-applied-statistics"><b>Annals of Applied Statistics</b></a>,
volume 18, September 2024.
<a href="https://github.com/stephenslab/poisson.mash.alpha">R package</a></p>

<p class=pub>
Youngseok Kim, Wei Wang, Peter Carbonetto and Matthew Stephens.
<a href="https://jmlr.org/papers/v25/22-0953.html">A flexible empirical
Bayes approach to multiple linear regression and connections with
penalized regression.</a>
<a href="https://jmlr.org"><b>Journal of Machine Learning Research</b></a>,
volume 25, 2024.
<a href="https://github.com/stephenslab/mr.ash.alpha/">R package</a></p>

<p class=pub>
Michihiro Takahama, Ashwini Patil, Gabriella Richey, Denis Cipurko,
Katherine Johnson, Peter Carbonetto, Madison Plaster, Surya Pandey,
Katerina Cheronis, Tatsuki Ueda, Adam Gruenbaum, Tadafumi Kawamoto,
Matthew Stephens and Nicolas Chevrier.
<a href="https://doi.org/10.1038/s41590-023-01722-8">
A pairwise cytokine code explains the organism-wide response to sepsis.</a>
<a href="https://www.nature.com/ni/"><b>Nature Immunology</b></a>,
volume 25, January 2024.</p>

<p class=pub>
Peter Carbonetto, Kaixuan Luo, Abhishek Sarkar, Anthony Hung, Karl Tayeb, 
Sebastian Pott and Matthew Stephens.
<a href="https://doi.org/10.1186/s13059-023-03067-9">GoM DE: 
interpreting structure in sequence count data with differential expression 
analysis allowing for grades of membership.</a>
<a href="https://genomebiology.biomedcentral.com"><b>Genome Biology</b></a>,
volume 24, October 2023.
<a href="https://github.com/stephenslab/fastTopics">R package</a></p>

<p class=pub>
Fabio Morgante, Peter Carbonetto, Gao Wang, Yuxin Zou, Abhishek Sarkar and
Matthew Stephens. 
<a href="https://doi.org/10.1371/journal.pgen.1010539">
A flexible empirical Bayes approach to multivariate multiple
regression, and its improved accuracy in predicting multi-tissue
gene expression from genotypes.</a>
<a href="http://www.plosgenetics.org"><b>PLoS Genetics</b></a>,
volume 19, July 2023.
<a href="https://github.com/stephenslab/mr.mash.alpha">R package</a> |
<a href="https://doi.org/10.5281/zenodo.8014360">code &amp; data</a></p>

<p class=pub>
Yuxin Zou, Peter Carbonetto, Gao Wang and Matthew Stephens.
<a href="https://doi.org/10.1371/journal.pgen.1010299">Fine-mapping 
from summary data with the &quot;Sum of Single Effects&quot; model.</a>
<a href="http://www.plosgenetics.org"><b>PLoS Genetics</b></a>,
volume 18, July 2022.
<a href="https://github.com/stephenslab/susieR">R package</a> |
<a href="https://doi.org/10.5281/zenodo.5611713">code &amp; data</a></p>

<p class=pub>
Selene M. Clay, Nathan Schoettler, Andrew M. Goldstein, Peter Carbonetto,
Matthew Dapas, Matthew C. Altman, Mario G. Rosasco, James E. Gern,
Daniel J. Jackson, Hae Kyung Im, Matthew Stephens, Dan L. Nicolae and
Carole Ober.
<a href="https://doi.org/10.1186/s13073-022-01058-2">Fine‑mapping
studies distinguish geneti risks for childhood‑ and adult‑onset asthma
in the HLA region.</a> 
<a href="https://genomemedicine.biomedcentral.com"><b>Genomic Medicine</b></a>
14:55, May 2022.</p>

<p class=pub>
Zhengrong Xing, Peter Carbonetto and Matthew Stephens.
<a href="https://jmlr.org/papers/v22/19-042.html">Flexible signal denoising
via flexible empirical Bayes shrinkage.</a> 
<a href="https://jmlr.org"><b>Journal of Machine Learning Research</b></a>
22(93), pages 1-28, June 2021.
<a href="https://github.com/stephenslab/smashr">R package</a> |
<a href="https://github.com/stephenslab/smash-paper">code &amp; data</a></p>

<p class=pub>
Gao Wang, Abhishek Sarkar, Peter Carbonetto and Matthew Stephens.
<a href="https://doi.org/10.1101/501114">A simple new approach to 
variable selection in regression, with application to genetic fine 
mapping.</a> 
<a href="https://rss.onlinelibrary.wiley.com/journal/14679868"><b>Journal 
of the Royal Statistical Society, Series B</b></a> volume 82,
pages 1273-1300, July 2020.
<a href="https://github.com/stephenslab/susieR">R package</a> |
<a href="https://github.com/stephenslab/susie-paper">code &amp; data</a></p>

<p class=pub>
Youngseok Kim, Peter Carbonetto, Matthew Stephens and Mihai Anitescu.
<a href="https://doi.org/10.1080/10618600.2019.1689985">A fast
algorithm for maximum likelihood estimation of mixture proportions
using sequential quadratic programming.</a> 
<a href="https://www.tandfonline.com/ucgs"><b>Journal of
Computational and Graphical Statistics</b></a> volume 29, pages 261-273,
2020. 
<a href="https://github.com/stephenslab/mixsqp">R package</a> |
<a href="https://github.com/stephenslab/mixsqp-paper">code &amp; data</a></p>

<p class=pub>
John D. Blischak, Peter Carbonetto and Matthew Stephens.
<a href="https://doi.org/10.12688/f1000research.20843.1">Creating and 
sharing reproducible research code the workflowr way.</a> 
<a href="https://f1000research.com"><b>F1000Research</b></a> 8:1749, 2019. 
<a href="https://github.com/jdblischak/workflowr">R package</a></p>

<p class=pub>
Lu&igrave;s Felipe Ventorim Ferr&atilde;o, Rom&aacute;rio Gava
Ferr&atilde;o, Maria Am&eacute;lia Gava Ferr&atilde;o, Aymbir&eacute;
Fonseca, Peter Carbonetto, Matthew Stephens and Antonio Augusto Franco
Garcia.
<a href="https://doi.org/10.1038/s41437-018-0105-y">Accurate
genomic prediction of Coffea canephora in multiple environments using
whole-genome statistical models.</a>
<a href="https://www.nature.com/hdy"><b>Heredity</b></a>
volume 122, pages 261-275, March 2019.
</p>

<p class=pub>
Sarah Urbut, Gao Wang, Peter Carbonetto and Matthew Stephens.
<a href="https://doi.org/10.1038/s41588-018-0268-8">Flexible
statistical methods for estimating and testing effects in genomic
studies with multiple conditions.</a> 
<a href="https://www.nature.com/hdy"><b>Nature Genetics</b></a>
volume 51, pages 187-195, January 2019.
<a href="https://github.com/stephenslab/mashr">R package</a> |
<a href="https://github.com/stephenslab/gtexresults">code &amp; data</a></p>

<p class=pub>
Ana I. Hernandez Cordero, Peter Carbonetto, Gioia Riboni Verri, Jennifer
Gregory, David Vandenbergh, Joe Gyekis, David Blizard and Arimantas
Lionikas. <a href="https://physoc.onlinelibrary.wiley.com/doi/10.14814/phy2.13561">Replication and discovery of musculoskeletal QTLs in LG/J and
SM/J advanced intercross lines.</a>
<a href="http://physreports.physiology.org"><b>Physiological Reports</b></a>
6: e13561, February 2018.
</p>

<p class=pub>
Peter Carbonetto, Xiang Zhou and Matthew Stephens.
<a href="https://arxiv.org/abs/1709.06597">varbvs: fast
variable selection for large-scale regression.</a> arXiv:1709.06597.
<a href="http://github.com/pcarbo/varbvs">code</a>
</p>

<p class=pub>
Eunjung Han*, Peter Carbonetto*, Ross Curtis, Yong Wang, Julie
Granka, Jake Byrnes, Keith Noto, Amir Kermany, Natalie Myres, Mathew
Barber, Kristin Rand, Shiya Song, Theodore Roman, Erin Battat, Eyal
Elyashiv, Harendra Guturu, Eurie Hong, Kenneth Chahine and Catherine
Ball. <a href="http://www.nature.com/articles/ncomms14238">Clustering of
770 thousand genomes reveals post-colonial
population structure of North America. 
<a href="http://www.nature.com/ncomms"><b>Nature Communications</b></a>
8: 14238, February 2017.
(* indicates shared first authorship)</p>

<p class=pub>
Laura Sittig, Peter Carbonetto, Kyle Engel, Kathleen Krauss,
Camila Barrios-Camacho and Abraham Palmer. 
<a href="http://www.cell.com/neuron/fulltext/S0896-6273(16)30505-0">Genetic 
background limits generalizability of genotype-phenotype relationships</a>. 
<a href="http://www.cell.com/neuron"><b>Neuron</b></a> volume 91, 
pages 1253-1259, September 2016. 
<a href="http://dx.doi.org/10.1016/j.neuron.2016.09.009">perspective</a> | 
<a href="http://github.com/pcarbo/neuron">code &amp; data</a>
</p>

<p class=pub>
Clarissa Parker*, Shyam Gopalakrishnan*, Peter Carbonetto*, Natalia 
Gonzales, Emily Leung, Yeonhee Park, Emmanuel Aryee, Joe Davis, David 
Blizard, Cheryl Ackert-Bicknell, Arimantas Lionikas, Jonathan Pritchard
and Abraham Palmer. 
<a href="http://www.nature.com/ng/journal/v48/n8/abs/ng.3609.html">Genome-wide 
association study of behavioral, physiological and gene expression traits 
in outbred CFW mice.</a> 
<a href="http://www.nature.com/ng"><b>Nature Genetics</b></a> 
volume 48, pages 919-926, August 2016. (* indicates shared first authorship) 
<a href="http://github.com/pcarbo/cfw">code</a> | 
<a href="http://dx.doi.org/10.5061/dryad.2rs41">data</a>
</p>

<p class=pub>
Laura Sittig, Peter Carbonetto, Kyle Engel, Kate Krauss and Abraham
Palmer. <a href="http://dx.doi.org/10.1111/gbb.12262">Integration of 
genome-wide association and extant brain expression QTL identifies 
candidate genes influencing prepulse inhibition in inbred F1 mice.</a> 
<a href="http://onlinelibrary.wiley.com/journal/10.1111/(ISSN)1601-183X"><b>Genes,
 Brain and Behavior</b></a>, volume 15, pages 260-270, February 2016.
<a href="http://github.com/pcarbo/ppi-f1-gwas">code and data</a></p>

<p class=pub>
Luisa Pallares, Peter Carbonetto, Shyam Gopalakrishnan, Clarissa Parker,
Cheryl Ackert-Bicknell, Abraham Palmer and Diethard Tautz.
<a href="http://dx.doi.org/10.1371/journal.pgen.1005607">Mapping of 
craniofacial traits in outbred mice identifies major developmental genes 
involved in shape determination.</a>
<a href="http://www.plosgenetics.org"><b>PLoS Genetics</b></a>,
volume 11, November 2015.
<a href="http://datadryad.org/resource/doi:10.5061/dryad.k543p">code 
and data</a></p>

<p class=pub>
Clarissa Parker*, Peter Carbonetto*, Greta Sokoloff, Yeonhee Park, 
Mark Abney and Abraham Palmer.
<a href="http://dx.doi.org/10.1534/genetics.114.167056">High-resolution 
genetic mapping of complex traits from a combined analysis of F2 and 
advanced intercross mice.</a> 
<a href="http://www.genetics.org"><b>Genetics</b></a>,
volume 198, pages 103-116, September 2014. 
(* indicates shared first authorship) 
<a href="http://github.com/pcarbo/lgsmfear">code</a></p>

<p class=pub>
Peter Carbonetto, Riyan Cheng, Joseph Gyekis, Clarissa Parker, 
David Blizard, Abraham Palmer and Arimantas Lionikas.
<a href="http://dx.doi.org/10.1152/physiolgenomics.00055.2014">Discovery 
and refinement of muscle weight QTLs in B6 x D2 advanced intercross mice.</a>
<a href="http://physiolgenomics.physiology.org"><b>Physiological 
Genomics</b></a>, volume 46, pages 571-582, August 2014.
<a href="http://github.com/pcarbo/b6d2muscle">code</a></p>

<p class=pub>
Peter Carbonetto and Matthew Stephens. 
<a href="http://dx.doi.org/10.1371/journal.pgen.1003770">Integrated 
enrichment analysis of variants and pathways in genome-wide association 
studies indicates central role for IL-2 signaling genes in type 1 diabetes, 
and cytokine signaling genes in Crohn's disease.</a>
<a href="http://www.plosgenetics.org"><b>PLoS Genetics</b></a>, volume 9,
October
2013. <a href="http://www.ncbi.nlm.nih.gov/pubmed/24098138">Pubmed</a> | 
<a href="http://www.hfsp.org/frontier-science/awardees-articles/enhancing-discovery-genetic-risk-factors-disease-using-online-pat">HFSP article</a> | 
<a href="http://github.com/pcarbo/bmapathway">code</a></p>

<p class=pub>
Xiang Zhou, Peter Carbonetto and Matthew Stephens. 
<a href="http://dx.doi.org/10.1371/journal.pgen.1003264">
Polygenic modeling with Bayesian sparse linear mixed models.</a> 
<a href="http://www.plosgenetics.org"><b>PLoS Genetics</b></a>, volume 9, 
February 2013.</p>

<p class=pub>
Peter Carbonetto and Matthew Stephens. <a href="bvs.pdf">Scalable 
variational inference for Bayesian variable selection in regression, 
and its accuracy in genetic association studies.</a> 
<a href="http://ba.stat.cmu.edu"><b>Bayesian Analysis</b></a>, 
volume 7, March 2012, pages 73-108. 
<a href="http://github.com/pcarbo/varbvs">code</a></p>

<p class=pub>
Matthew Hoffman, Peter Carbonetto, Nando de Freitas and Arnaud Doucet. 
<a href="http://www.cs.ubc.ca/&#126;hoffmanm/pub/2009-nips-smdp.pdf">Inference
strategies for solving SMDPs.</a> NIPS Workshop on Probabilistic
Approaches for Robotics and Control, December 2009.</p>

<p class=pub>
Peter Carbonetto, Matthew King and Firas Hamze. <a href="smcopt.pdf">A
stochastic approximation method for inference in probabilistic
graphical models.</a> Neural Information Processing Systems 23,
December 2009.</p>

<p class=pub>
Peter Carbonetto, Mark Schmidt and Nando de Freitas. 
<a href="stocip.pdf">An interior-point stochastic approximation method 
and an L1-regularized delta rule.</a> Neural Information Processing 
Systems 22, December 2008. (<b>Note:</b> the proof of asymptotic 
convergence that was originally published as an appendix in the original
paper has a major flaw; the convergence proof remains an open question.)
<a href="stocip-slides.pdf">slides</a> | 
<a href="#code">code</a></p>

<p class=pub>
Peter Carbonetto, Gyuri Dork&ograve;, Cordelia Schmid, Hendrik
K&uuml;ck and and Nando de Freitas. <a href="ijcvss.pdf">Learning to
recognize objects with little supervision.</a> 
<a href="http://www.springerlink.com/content/100272"><b>International Journal
of Computer Vision</b></a>, volume 77, May 2008, pages 219-237.</p>

<p class=pub>
Peter Carbonetto and Nando de Freitas. <a href="cmf.pdf">Conditional
mean field.</a> Neural Information Processing Systems 19, 
December 2006, pages 201-208.</p>

<p class=pub>
Peter Carbonetto, Jacek Kisynski, Nando de Freitas and David Poole.
<a href="npblog.pdf">Nonparametric Bayesian Logic.</a> 
21st Conference on Uncertainty in Artificial Intelligence, July 2005, 
pages 85-93. <i>This revision corrects a mistake in Fig. 5.</i></p>

<p class=pub>
Peter Carbonetto, Gyuri Dork&ograve; and Cordelia Schmid. 
<a href="techreport.pdf">Bayesian learning for weakly supervised 
object classification.</a> Technical Report, INRIA Rh&ocirc;ne-Alpes, 
July 2004.
</p>

<p class=pub>
Peter Carbonetto, Nando de Freitas and Kobus Barnard. 
<a href="mrftrans.pdf">A Statistical Model for General Contextual Object 
Recognition.</a> 8th European Conference on Computer Vision, May 
2004, part I, pages 350-362.<sup><a href="#note">1</a></sup> 
</p>

<p class=pub>
Hendrik K&uuml;ck, Peter Carbonetto and Nando de Freitas.
<a href="semisup.pdf">A Constrained Semi-Supervised Learning Approach to
Data Association</a>. 8th European Conference on Computer Vision, May 
2004, part III, pages 1-12.<sup><a href="#note">1</a></sup>
</p>

<p class=pub>
Peter Carbonetto and Nando de Freitas. <a href="robotrans.pdf">Why 
can't Jos&eacute; read? The problem of learning semantic associations
in a robot environment</a>. Human Language Technology Conference
Workshop on Learning Word Meaning from Non-Linguistic Data, June 
2003.</p>

<p class=pub>
Peter Carbonetto, Nando de Freitas, Paul Gustafson and Natalie Thompson.
<a href="shrinkage.pdf">Bayesian feature weighting for unsupervised
learning, with application to object recognition</a>. Workshop on
Artificial Intelligence and Statistics, January 2003.
</p>

<a name="patents"></a><h1>Patents</h1>

<a class=pub>Eunjung Han, Ross E. Curtis and Peter Carbonetto. 
<a href="https://patents.google.com/patent/US9940433B2">Discovering 
population structure from patterns of identity-by-descent</a>,
April 10, 2018, U.S. Patent 9,940,433.
</a>

<a name="theses"></a><h1>Theses</h1>

<p class=pub><a href="thesis.pdf">New probabilistic inference algorithms that 
harness the strengths of variational and Monte Carlo methods</a>. 
Ph.D. thesis, University of British Columbia, August 2009.</p>

<p class=pub><a href="masters.pdf">Unsupervised Statistical Models for
General Object Recognition</a>. Masters thesis, University of
British Columbia, August 2003.</p>

<a name="code"></a><h1>Code</h1>

<p class=pub>
<a class="code" href="http://github.com/Ancestry/admixture">Admixture.</a> 
A simple EM implementation of the 
<a href="http://www.genetics.ucla.edu/software/admixture/">ADMIXTURE</a> 
model in R, plus extensions.</p>

<p class=pub>
<a class="code"
   href="http://github.com/pcarbo/Variational-inference-for-Bayesian-variable-selection">Variational
  inference for Bayesian variable selection in MATLAB and R.</a> 
Companion code to my Bayesian Analysis (2012)
paper. Includes routines for computing variational estimates of posterior
statistics, and demonstrates how to run the full variational inference
procedure for Bayesian variable selection in linear and logistic regression.  
</p>

<p class=pub>
<a class="code" href="stocip-matlab.tar.gz">MATLAB code for on-line L1 
regularization.</a> Companion code to my research paper appearing at the 
2008 NIPS conference (see below for <a href="#data">data</a>). 
Includes MATLAB functions for learning 
linear regressors and classifiers subject to L1 regularization, which 
acts as a form of feature selection. The linear regression is also known 
in the statistics community as the LASSO. The software package includes 
implementations of both batch learning and on-line learning, when the 
model parameters are rapidly adjusted at each iteration by looking at 
only a single training example. 
<em>This software is licensed under the 
<a href="http://creativecommons.org/licenses/GPL/2.0/">CC-GNU GPL</a> 
version 2.0 or later.</em>
</p>

<p class=pub>
<a class=code href="ssmcmcmatlab.tar.gz">Semi-supervised
classification using a Bayesian kernel machine and data association
constraints.</a> Matlab implementation of the MCMC algorithms for
simulating the Bayesian data association models described in the <a
href="#pubs">ECCV 2004 paper</a> and the <a href="#pubs">INRIA tech
report</a> (the data association model with hard group constraints),
and <a href="http://www.cs.ubc.ca/spider/nando/publications.html">Learning
to classify individuals based on group statistics</a> by Kuck and de
Freitas (data association with group statistics). <b>For a much more
stable implementation in C, go <a
href="objrecls/index.html#ssmcmc">here</a>.</b></p>

<p class=pub>
<a class=code href="gbp.tar.gz">Gaussian belief propagation.</a>
Matlab code for running belief propagation on Gaussian Markov random
fields.</p>

<p class=pub>
<a class=code href="imagetrans.tar.gz">Image Translation.</a>
Matlab package for generic object recognition using statistical
translation models. See my <a href="#thesis">Masters thesis</a> for
more information.</p>

<p class=pub>
<a class=code href="feature-weighting.tar.gz">Feature Weighting using
Shrinkage Priors.</a> Matlab code for running EM on a mixture of
Gaussians with Bayesian feature weighting priors. Used for the paper
<a href="#pubs">Bayesian feature weighting for unsupervised 
learning.</a></p>

<p class=pub>

<a class=code href="dispatch.tar.gz">Multiple dispatch.</a> An
implementation of multiple dispatch in Java using the 
<a href="http://www.cs.ubc.ca/labs/spl/projects/elide/">ELIDE</a>
framework. See <a href="cs539s.pdf">here</a> for the project 
report.</a>

<a name="data"></a><h1>Data</h1>

<p class=pub>
<a class=code href="trec2005.tar.gz">TREC2005.</a> Spam filtering data
in MATLAB format. Used to evaluate my on-line logistic regression learning 
algorithm in the paper <a href="#pubs">An interior-point stochastic 
approximation method and an L1-regularized delta rule</a>. This data set
was originally created by 
<a href="http://plg.uwaterloo.ca/~gvcormac">Gordon Cormack</a> and 
<a href="http://plg1.cs.uwaterloo.ca/~trlynam">Thomas Lynam</a> as part of 
the <a href="http://plg1.cs.uwaterloo.ca/~trlynam/spamjig">2005 TREC Spam 
Filter Evaluation Tool Kit</a>, and contains data from 92,189 emails. The 
open source software <a href="http://spambayes.sourceforge.net">SpamBayes</a>
 was used to extract features from the emails. By downloading and using 
this data, <em>you 
accept the <a href="http://plg.uwaterloo.ca/~gvcormac/treccorpus">terms 
of agreement</a> for use of the 2005 TREC public spam corpus.</em>

<p class=pub>
<a class=code href="corel.tar.gz">Corel.</a> Object recognition data
used for my <a href="#thesis">Masters thesis</a> and the paper 
<a href="#pubs">A Statistical Model for General Contextual Object
Recognition.</a> Contains manual segmentations for evaluation and
extracted featres. The <a href="#code">Image Translation</a> package
contains code for reading the data into Matlab.</p>

<p class=pub>
<a class=code href="robomedia.tar.gz">Robomedia.</a> 
Object recognition data used for the <a href="#pubs">Why can't Jos&eacute; 
read?</a> paper. Contains manual segmentations for evaluation and extracted 
featres. The <a href="#code">Image Translation</a> package contains code 
for reading the data into Matlab.</p>

<p class=pub>
<a class=code href="viola-traindata.tar.gz">Face detection.</a>
Training data for robust object detection using the AdaBoost
algorithm, as formalized by Viola and Jones. Includes Matlab code for
reading the data. The project report is available 
<a href="cs530project.pdf">here</a>.</p>

<a name="other"></a><h1>Other work</h1>

<p class="pub"><a href="http://www.pardiso-project.org">MATLAB interface 
for PARDISO.</a> PARDISO is a 
publicly available software library for solving large, sparse linear 
systems. It is particularly useful as a subroutine for interior-point
methods. I designed a small interface so that the PARDISO solver is easily
  incorporated into your MATLAB programs.</p>

<p class="pub"><a href="http://github.com/pcarbo/lbfgsb-matlab">MATLAB 
class for limited-memory BFGS.</a> 
This little MATLAB class I wrote encapsulates
all the functionality of limited-memory quasi-Newton methods. It is
particularly well-suited for solving constrained optimization
problems; I illustrate how it it is used within a primal-dual
interior-point method for solving a constrained optimization problem
that arises in maximum likelihood estimation. See <a
href="http://github.com/pcarbo/lbfgsb-matlab">here</a> 
for more details on installing and using this software.</p>

<p class="pub"><a href="lp.pdf">Intuition behind primal-dual
interior-point methods for linear and quadratic programming.</a> I'm
quite aware of the fact that there are probably a hundred textbooks
published every year that contain an introduction to linear
programming, and there are many introductory presentations on
interior-point methods. But I find they are all lacking in providing
the key intuition. So I've written a short 7-page document which I'm
confident fills a tiny bit of the void.</p>

<p class="pub"><a href="convexprog.html">MATLAB code for solving constrained,
convex programs</a>. I wrote a simple, easy-to-use MATLAB function
for minimizing a convex objective subject to convex inequality constraints.
It uses a primal-dual interior-point method with a suitable merit function
for ensuring global convergence (which is useful when it is not 
desirable to compute the Newton step using the full Hessian of the 
objective).</p>

<p class=pub><a href="socpprog.html">MATLAB code for second-order cone
programming</a>. I also implemented a simple primal-dual interior-point
method in MATLAB for solving second-order cone programs. At each 
iteration, the solver follows the Newton search direction and makes
sure that the iterates remain feasible (they satisfy all the inequality
constraints).</p>

<p class=pub><a
href="https://projects.coin-or.org/Ipopt">MATLAB interface for IPOPT.</a> 
IPOPT is a fantastic, new open source
software package written in C++ for solving optimization problems with
nonlinear objectives and subject to nonlinear constraints. IPOPT is
short for Interior Point Optimizer. I've developed an interface so
that IPOPT can be easily called from the MATLAB programming
environment. You can download the current version of IPOPT from the <a
href="https://projects.coin-or.org/Ipopt">project website</a>.</p>

<p class=pub><a href="ldpc.pdf">Notes on probabilistic decoding of parity
check matrices.</a> A review of the basic concepts behind low-density parity
check codes, and how to come up with a simple and reasonable method for
probabilistic decoding. Assumes some familiarity with some ideas in 
statistical machine learning concepts and optimization.</p>

<p class=pub><a href="lbfgsb-for-matlab.html">A MATLAB interface for 
L-BFGS-B</a>, a solver for bound-constrained nonlinear optimization problems
that uses quasi-Newton updates with a limited-memory approximation to the 
Hessian.</p>

<p class=pub><a href="dual.pdf">A non-rigorous derivation of a
variational upper bound on the log-partition function in eight
parts.</a> This is a brief expos&eacute; of Martin Wainwright's
derivation of a convex alternative to generalized belief propagation
(resulting in the so-called tree-reweighted belief propagation
algorithm). The intent is to present the main mathematical steps in
the derivation while keeping the presentation as &quot;light&quot; as
possible.</p>

<p class=pub><a href="ipopt-macosx/index.html">Installing IPOPT on 
Mac OS X.</a> Some of my experiences.</p>

<p class=pub><a href="mex-tutorial/index.html">Creating, compiling and linking
MATLAB executables (MEX files).</a> A tutorial.</p>

<p class=pub><a href="snooker.pdf">A Lesson in measure theory and change of
variables.</a> A technical note illustrating and explaining the
subtleties in deriving a correct kernel for the snooker move used in
population Monte Carlo.</p>

<p class=pub>Project webpage for <a href="objrecls/index.html">Learning 
to recognize objects with little supervision.</a></p>

<p class=pub><a href="macnexstar.html">How to partition and format an external 
hard drive for Mac OS X.</a></p>

<a name="note"></a><h1>Note</h1>
<p class=pub>
<b><sup>1</sup></b> 
&copy; Springer-Verlag. Published in the Springer-Verlag
<a href="http://www.springer.de/comp/lncs/index.html">Lecture Notes in
Computer Science</a> series.
</p>

<p><script type="text/javascript" language="JavaScript">
var site="s13miamisquirrel"</script> 
<script type="text/javascript" language="JavaScript1.2"
src="http://s13.sitemeter.com/js/counter.js?site=s13miamisquirrel"> 
</script> <noscript> 
<a href="http://s13.sitemeter.com/stats.asp?site=s13miamisquirrel" 
target="_top"> 
<img src="http://s13.sitemeter.com/meter.asp?site=s13miamisquirrel" 
alt="Site Meter" border=0></a> </noscript> 
</p>

</td></tr></table>

</center>
</body>
</html>
